cms/djangoapps/contentstore/views/tests/test_import_export.py

```diff
@@ -1021,3 +1021,97 @@ def test_library_content_on_course_export_import(self, publish_item):
             dest_course.location,
             publish_item
         )
+
+
+@ddt.ddt
+@override_settings(CONTENTSTORE=TEST_DATA_CONTENTSTORE)
+class TestCourseExportImportProblem(CourseTestCase):
+    """
+    Tests for importing after exporting the course containing problem with pre tags from XML.
+    """
+    def setUp(self):
+        super(TestCourseExportImportProblem, self).setUp()
+        self.export_dir = tempfile.mkdtemp()
+        self.source_course = CourseFactory.create(default_store=ModuleStoreEnum.Type.split)
+        self.addCleanup(shutil.rmtree, self.export_dir, ignore_errors=True)
+
+    def _setup_source_course_with_problem_content(self, publish_item=False):
+        """
+        Sets up course with problem content.
+        """
+        chapter = ItemFactory.create(
+            parent_location=self.source_course.location,
+            category='chapter',
+            display_name='Test Section'
+        )
+        sequential = ItemFactory.create(
+            parent_location=chapter.location,
+            category='sequential',
+            display_name='Test Sequential'
+        )
+        vertical = ItemFactory.create(
+            category='vertical',
+            parent_location=sequential.location,
+            display_name='Test Unit'
+        )
+
+        ItemFactory.create(
+            parent_location=vertical.location,
+            category='problem',
+            display_name='Test Problem',
+            publish_item=publish_item,
+            data='<problem><pre><code>x=10</code></pre><multiplechoiceresponse></multiplechoiceresponse></problem>',
+        )
+
+    def get_problem_content(self, block_location):
+        """
+        Get problem content after exported course in imported back in
+        """
+        if block_location.block_type == 'problem':
+            return self.store.get_item(block_location).data
+
+        return self.get_problem_content(self.store.get_item(block_location).children[0])
+
+    def assert_problem_definition(self, dest_course_location):
```

Comment:
`dest_course_location` => `course_location `

Reply:
you can avoid the recursive call by
`problem_location = course_location.make_usage_key('problem', '{problem_id}')`
and then
`problem_after_export = self.store.get_item(problem_location).data`